### PR TITLE
create guide to set LSP8 NFT metadata

### PR DESCRIPTION
**Pull request by CJ42**

### CODE REVIEW

File: docs/learn/dapp-developer/set-nft-metadata.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+- [`ethers.js`](https://github.com/ethers-io/ethers.js/)
+- [`@lukso/lsp-smart-contracts`](https://github.com/lukso-network/lsp-smart-contracts/)
+
+```shell
+npm install ethers @lukso/lsp-smart-contracts
+```
+
+## Imports and constants
+
+At this point, we are preparing the `LSP8IdentifiableDigitalAsset` contract to interact with it. We will construct an instance of the contract using its _ABI_ and the _contract address_.
```
Comment (Hugoo) [inline docs/learn/dapp-developer/set-nft-metadata.md:26]: Maybe something more straight to the point like: ?

```suggestion
Import `ethers`, the [`LSP8IdentifiableDigitalAsset` ABI](relative link to https://docs.lukso.tech/contracts/contracts/LSP8IdentifiableDigitalAsset/) from [@lukso/lsp-smart-contracts](HERE RELATIVE LINK to https://docs.lukso.tech/tools/lsp-smart-contracts/getting-started) and create an instance of this contract with the `lsp8ContractAddress`.
```

File: docs/learn/dapp-developer/set-nft-metadata.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+const lsp8Contract = new ethers.Contract(lsp8ContractAddress, LSP8IdentifiableDigitalAsset.abi, signer);
+```
+
+### Send the transaction to set metadata for a tokenId
+
+The last step is to set the metadata for a specific `tokenId`. You can then send the transaction to set the metadata for the given NFT represented by its `tokenId`.
+
+Here we will use the following parameters as examples:
+
+- `tokenId`: `0x0000000000000000000000000000000000000000000000000000000000000001`
```
Comment (Hugoo) [inline docs/learn/dapp-developer/set-nft-metadata.md:56]: does `tokenId` support hex or decimal?

Where can i find the specs about this tokenId value?

Might be nice to add more info/links here.

For token 10 shall I write:

```
0x0000000000000000000000000000000000000000000000000000000000000010
```

or 

```
0x000000000000000000000000000000000000000000000000000000000000000A
```

?

File: docs/learn/dapp-developer/set-nft-metadata.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+- `dataKey`: [`LSP4Metadata`](../../standards/tokens/LSP4-Digital-Asset-Metadata.md#lsp4metadata)
+- `metadataValue`: some placeholder value as a .
```
Comment (Hugoo) [inline docs/learn/dapp-developer/set-nft-metadata.md:58]: "value as a ." ? 🤔

File: docs/learn/dapp-developer/set-nft-metadata.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+```javascript
+import {ERC725YDataKeys} from "@lukso/lsp-smart-contracts";
+
+// change this with the actual tokenId to update metadata for
+const tokenId = "0x0000000000000000000000000000000000000000000000000000000000000001";
+
+// change this with the actual VerifiableURI of the metadata uploaded on IPFS
```
Comment (Hugoo) [inline docs/learn/dapp-developer/set-nft-metadata.md:66]: I think you should explain or put a link to VerifiableURI before this code snippet. Readers will be surprised to see VerifiableURI here they will be like wtf?

File: docs/learn/dapp-developer/set-nft-metadata.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+const metadataValue = "0x..."
```
Comment (Hugoo) [inline docs/learn/dapp-developer/set-nft-metadata.md:67]: cf comment above - we don't rly know what this is

File: docs/learn/dapp-developer/set-nft-metadata.md (hunk 4 above)
Comment (Hugoo) [reply docs/learn/dapp-developer/set-nft-metadata.md:66]: It doesnt need to be on IPFS, so i suggest to not write it like this.

"uploaded somewhere (S3, IPFS, etc.)" something like that

File: docs/learn/dapp-developer/set-nft-metadata.md (hunk 5 above)
Comment (Hugoo) [reply docs/learn/dapp-developer/set-nft-metadata.md:67]: Maybe you need to explain how to encode to VerifiableURI

File: docs/learn/dapp-developer/set-nft-metadata.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+import LSP8IdentifiableDigitalAsset from '@lukso/lsp-smart-contracts/artifacts/LSP8IdentifiableDigitalAsset.json';
+import {ERC725YDataKeys} from "@lukso/lsp-smart-contracts";
+
+const privateKey = '0x...';
+const lsp8ContractAddress = '0x...';
+
+async function main() {
+
+    const provider = new ethers.JsonRpcProvider(
+        'https://rpc.testnet.lukso.network',
```
Comment (Hugoo) [inline docs/learn/dapp-developer/set-nft-metadata.md:90]: Maybe put this in a constant above:

```js
// For networks info, check: https://docs.lukso.tech/networks/testnet/parameters
const LUKSO_TESTNET_RPC_URL = 'https://rpc.testnet.lukso.network'
```

File: docs/learn/dapp-developer/set-nft-metadata.md (hunk 2 above)
Comment (CJ42) [reply docs/learn/dapp-developer/set-nft-metadata.md:56]: It is always a `bytes32` as hex value. We can link to the LSP8 standard page in the docs.

Comment (Hugoo) [reply docs/learn/dapp-developer/set-nft-metadata.md:74]: I suggest to add it right here. It doesn't hurt

```
Note that the tokenId is a bytes32 value (hex)
```

**Review by Hugoo — APPROVED** [review]
great!!

**Comment by fhildeb** [comment]
LGTM

**Pull request merged**